docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,570 @@
+by design.
+Localnet topology requires the net-conf-network-name to match the OVN bridge-mapping network-name on the node.
+In case [Kubernetes-nmstate](https://nmstate.io/kubernetes-nmstate/) is used, the NAD `spec.config.name` has to match the `NodeNetworkConfigurationPolicy`
+`spec.desiredState.ovn.bridge-mappings` name:
+```yaml
+spec:
+  desiredState:
+      ovn:
+        bridge-mappings:
+        - localnet: physnet  <---- has to match the NAD config.spec.name. OR 
+                                   the NAD config.spec.physicalNetworkName
+          bridge: br-ex <--------- OVS switch
+```
+* To overcome this, and avoid exposing the net-conf-network-name in the CUDN CRD spec a new field should be introduced.
+  The new field should allow users pointing to the bridge-mapping network-name they defined in the node.
+  The field should be translated to the CNI physicalNetworkName field.
+
+##### MTU
+By default OVN-K sets the pod interface with MTU 1400, unless otherwise specified.
+
+For localnet topology this is not optimal because the host physical network-interface usually has default MTU 1500.
+In case no MTU is specified the pod interface connected to the host localnet is configured with MTU 1400 resulting
+in bad performance and bad UX requiring troubleshooting the MTU alignment across the stack.
+
+As long OVN-K does not set default MTU (1500) for localnet topology network,
+the controller should specify MTU to be 1500 explicitly in the NAD, if no MTU was specified in the CUDN spec.
+
+#### Workflow Description
+
+The CUDN CRD controller should be changed accordingly to support localnet topology.
+It should validate localnet topology configuration and generate corresponding NADs for localnet as other topologies (Layer2 & Layer3)
+in the selected namespaces.
+
+As of today the CUDN CRD controller does not allow mutating `spec.network`.
+
+OVN-Kubernetes support NAD spec mutations on day 2.
+For the new network configuration (defined in the spec) to take place, the user must restart the workloads.
+
+In order to comply with NAD spec mutation support and provide better UX,
+the CUDN CRD controller should allow `spec.network` mutations for localnet topologies.
+Enable changing at least the following fields: `MTU`, `VLAN`, `excludeSubnets` and `physicalNetworkName`.
+
+In any order:
+- The user configures localnet bridge mapping on the nodes, e.g.: using NNCP.
+- Create CUDN, specifying the bridge-mapping network name in the spec.
+
+#### Generating the NAD
+##### OVS bridge-mapping’s network-name
+Introduce attribute to point ovs-bridge bridge-mapping network-name.
+This attribute name should be translated to the CNI “physicalNetworkName” attribute.
+
+Proposal the CUDN spec field name:  “physicalNetworkName”.
+
+##### MTU
+Should be translated to the CNI “mtu” attribute.
+If not specified the controller should specify default MTU (1500) in the NAD spec.config.
```

Comment:
may also need some cleanup with the rewording of MTU

Reply:
Done